engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -627,4 +626,110 @@ public function disableEntities($owner_guid) {
+			return false;
+		}
+
+		// first check if we can delete this entity
+		// NOTE: in Elgg <= 1.10.3 this was after the delete event,
+		// which could potentially remove some content if the user didn't have access
+		if (!$entity->canDelete()) {
+			return false;
+		}
+
+		if (!_elgg_services()->hooks->getEvents()->triggerBefore('delete', $entity->type, $entity)) {
+			return false;
+		}
+
+		// now trigger an event to let others know this entity is about to be deleted
+		// so they can prevent it or take their own actions
+		if (!_elgg_services()->hooks->getEvents()->trigger('delete', $entity->type, $entity)) {
+			return false;
+		}
+
+		if ($entity instanceof ElggUser) {
+			// ban to prevent using the site during delete
+			$entity->ban();
+		}
+
+		// Delete contained owned and otherwise releated objects (depth first)
+		if ($recursive) {
+			// Temporarily overriding access controls
+			$entity_disable_override = access_get_show_hidden_status();
+			access_show_hidden_entities(true);
+			$ia = elgg_set_ignore_access(true);
+
+			// @todo there was logic in the original code that ignored
+			// entities with owner or container guids of themselves.
+			// this should probably be prevented in \ElggEntity instead of checked for here
+			$base_options = [
+				'wheres' => [
+					"e.guid != $guid",
+				],
+				'limit' => false,
+			];
+
+			foreach (['owner_guid', 'container_guid'] as $db_column) {
+				$options = $base_options;
+				$options[$db_column] = $guid;
+
+				$batch = new \ElggBatch('elgg_get_entities', $options);
+				$batch->setIncrementOffset(false);
+
+				/* @var $e \ElggEntity */
+				foreach ($batch as $e) {
+					$this->delete($e, true);
+				}
+			}
+
+			access_show_hidden_entities($entity_disable_override);
+			elgg_set_ignore_access($ia);
+		}
+
+		$entity_disable_override = access_get_show_hidden_status();
+		access_show_hidden_entities(true);
+		$ia = elgg_set_ignore_access(true);
+
+		// Now delete the entity itself
+		$entity->deleteMetadata();
+		$entity->deleteAnnotations();
+		$entity->deleteOwnedAnnotations();
+		$entity->deleteRelationships();
+		$entity->deleteAccessCollectionMemberships();
+		$entity->deleteOwnedAccessCollections();
+
+		access_show_hidden_entities($entity_disable_override);
+		elgg_set_ignore_access($ia);
+
+		elgg_delete_river(['subject_guid' => $guid, 'limit' => false]);
+		elgg_delete_river(['object_guid' => $guid, 'limit' => false]);
+		elgg_delete_river(['target_guid' => $guid, 'limit' => false]);
+
+		remove_all_private_settings($guid);
+
+		$dir = new \Elgg\EntityDirLocator($guid);
+		$file_path = _elgg_config()->dataroot . $dir;
+		delete_directory($file_path);
+
+		_elgg_invalidate_cache_for_entity($guid);
+
+		$qb = Delete::fromTable('entities');
+		$qb->where($qb->compare('guid', '=', $guid, ELGG_VALUE_INTEGER));
+
+		$deleted = $this->db->deleteData($qb);
+
+		_elgg_services()->hooks->getEvents()->triggerAfter('delete', $entity->type, $entity);
+
+		return (bool) $deleted;
```

Comment:
When would we want to return false?

Reply:
I am not sure. Just moved this from ElggEntity